Client/mods/deathmatch/logic/luadefs/CLuaVectorGraphicDefs.cpp

```diff
@@ -0,0 +1,133 @@
+        }
+        else if (vecSize.fY < 0)
+        {
+            argStream.SetCustomError("Height is smaller than 0", "Invalid parameter");
+        }
+        else if (vecSize.fX == 0 || vecSize.fY == 0)
+        {
+            m_pScriptDebugging->LogWarning(luaVM, "A vector graphic must be at least 1x1 in size. This warning may be an error in future versions.");
+        }
+    }
+
+    if (!argStream.HasErrors())
+    {
+        CLuaMain* pLuaMain = m_pLuaManager->GetVirtualMachine(luaVM);
+        if (pLuaMain)
+        {
+            CResource* pParentResource = pLuaMain->GetResource();
+
+            CClientVectorGraphic* pVectorGraphic =
+                g_pClientGame->GetManager()->GetRenderElementManager()->CreateVectorGraphic((int)vecSize.fX, (int)vecSize.fY);
+
+            if (pVectorGraphic)
+            {
+                // Make it a child of the resource's file root ** CHECK  Should parent be pFileResource, and element added to pParentResource's ElementGroup? **
+                pVectorGraphic->SetParent(pParentResource->GetResourceDynamicEntity());
+
+                // Set our owner resource
+                pVectorGraphic->SetResource(pParentResource);
+
+                if (!path.empty())
+                {
+                    // Try and load from the filepath or svg data (if provided)
+                    std::string strPath;
+                    if (CResourceManager::ParseResourcePathInput(path, pParentResource, &strPath) && FileExists(strPath))
+                    {
+                        if (!pVectorGraphic->LoadFromFile(strPath))
+                        {
+                            pVectorGraphic->Destroy();
+                            delete pVectorGraphic;
+                            pVectorGraphic = nullptr;
+
+                            m_pScriptDebugging->LogError(luaVM, "Unable to load SVG data");
+                            lua_pushboolean(luaVM, false);
+                            return 1;
+                        }
+
+                    }
+                    else
+                    {
+                        pVectorGraphic->Destroy();
+                        delete pVectorGraphic;
+                        pVectorGraphic = nullptr;
+
+                        m_pScriptDebugging->LogError(luaVM, SString("Unable to load SVG file [%s]", path.c_str()).c_str());
+                        lua_pushboolean(luaVM, false);
+                        return 1;
+                    }
+                        
+                }
+
+            }
+            lua_pushelement(luaVM, pVectorGraphic);
+            return 1;
+        }
+    }
+    else
+        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
+
+    lua_pushboolean(luaVM, false);
+    return 1;
+}
+
+std::variant<bool, int> CLuaVectorGraphicDefs::SVGAddRect(CClientVectorGraphic* pVectorGraphic, std::variant<float, std::string> x, std::variant<float, std::string> y,
```

Comment:
You might want to add some general type `using` declaration for `std::variant<float, std::string>` to keep this a bit more readable